app/code/core/Mage/Catalog/Model/Resource/Abstract.php

```diff
@@ -584,8 +584,12 @@ public function getAttributeRawValue($entityId, $attribute, $store)
     {
         if (!$entityId || empty($attribute)) {
             return false;
-        }
+	}
+
+        $wasArray = true;
```

Comment:
To better maintain BC I think `$wasArray` should also be false if the size of the `$attribute` array was 1. In this case perhaps it should be renamed to something like `$returnArray`.

Reply:
You are right, I didn't saw any code like that in Magento, but some 3rd party module maybe relying on it. Fixed that also.